test/mocha/10-restrictions.js

```diff
@@ -358,10 +393,63 @@ describe('restrictions', function() {
     let err;
     try {
       // try getting the removed restriction, this should throw a NotFoundError
-      getRestriction2 = await restrictions.get({
+      getRestriction2 = await restrictions.get({id});
+    } catch(e) {
+      err = e;
+    }
+    should.not.exist(getRestriction2);
+    should.exist(err);
+    err.name.should.equal('NotFoundError');
+    err.message.should.equal('Restriction not found.');
+  });
+
+  it('should remove a restriction from the database by id', async function() {
+    // create restriction
+    const id = await generateId();
+    const actualRestriction = await restrictions.insert({
+      restriction: {
+        id,
         zone: ZONES.ONE,
-        resource: RESOURCES.MANGO
-      });
+        resource: RESOURCES.MANGO,
+        method: 'limitOverDuration',
+        methodOptions: {
+          limit: 1,
+          duration: 'P30D'
+        }
+      }
+    });
+    const expectedRestriction = {
+      id,
+      zone: ZONES.ONE,
+      resource: RESOURCES.MANGO,
+      method: 'limitOverDuration',
+      methodOptions: {
+        limit: 1,
+        duration: 'P30D'
+      }
+    };
+    should.exist(actualRestriction);
+    should.exist(actualRestriction.meta);
+    should.exist(actualRestriction.restriction);
+    actualRestriction.restriction.should.deep.equal(expectedRestriction);
+
+    const getRestrictions = await restrictions.getAll({
```

Comment:
I suggest using the `get` API for this sort of confirmation that the record exists in the database since you know exactly what you're looking for.  Then you don't need to be concerned about the length of the `restrictions` array etc.

It's unclear to me by this diff if there is now adequate testing for both the `getAll` API and the `get` API.  Are there tests that demonstrate that multiple restrictions are returned properly?

Reply:
I have changed it to just use `get` as we are only interested in confirming the record exists. 

I am creating a separate test that just tests `getAll`.